libsawtooth/src/receipt/store/mod.rs

```diff
@@ -87,3 +92,28 @@ pub trait ReceiptStore: Sync + Send {
         id: Option<String>,
     ) -> Result<Box<dyn Iterator<Item = TransactionReceipt>>, ReceiptStoreError>;
 }
+
+/// Trait used by the LMDB `ReceiptStore` implementation to convert from bytes back
+/// to the native type.
+#[cfg(feature = "transaction-receipt-store-lmdb")]
+pub trait FromBytes: Sized {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, String>;
```

Comment:
Should this return a proper error, like InternalError?

Reply:
Per Shawn's comment I removed the AsBytes and FromBytes traits and implementations